monkey_island/cc/services/report.py

```diff
@@ -0,0 +1,422 @@
+        if exploiter_type == 'SSHExploiter':
+            return ReportService.process_ssh_exploit(exploit)
+        if exploiter_type == 'RdpExploiter':
+            return ReportService.process_rdp_exploit(exploit)
+        if exploiter_type == 'SambaCryExploiter':
+            return ReportService.process_sambacry_exploit(exploit)
+        if exploiter_type == 'ElasticGroovyExploiter':
+            return ReportService.process_elastic_exploit(exploit)
+        if exploiter_type == 'Ms08_067_Exploiter':
+            return ReportService.process_conficker_exploit(exploit)
+        if exploiter_type == 'ShellShockExploiter':
+            return ReportService.process_shellshock_exploit(exploit)
+
+    @staticmethod
+    def get_exploits():
+        exploits = []
+        for exploit in mongo.db.telemetry.find({'telem_type': 'exploit', 'data.result': True}):
+            new_exploit = ReportService.process_exploit(exploit)
+            if new_exploit not in exploits:
+                exploits.append(new_exploit)
+        return exploits
+
+    @staticmethod
+    def get_monkey_subnets(monkey_guid):
+        network_info = mongo.db.telemetry.find_one(
+            {'telem_type': 'system_info_collection', 'monkey_guid': monkey_guid},
+            {'data.network_info.networks': 1}
+        )
+        if network_info is None:
+            return []
+
+        return \
+            [
+                ipaddress.ip_interface(unicode(network['addr'] + '/' + network['netmask'])).network
+                for network in network_info['data']['network_info']['networks']
+            ]
+
+    @staticmethod
+    def get_cross_segment_issues():
+        issues = []
+        island_ips = local_ip_addresses()
+        for monkey in mongo.db.monkey.find({'tunnel': {'$exists': False}}, {'tunnel': 1, 'guid': 1, 'hostname': 1}):
+            found_good_ip = False
+            monkey_subnets = ReportService.get_monkey_subnets(monkey['guid'])
+            for subnet in monkey_subnets:
+                for ip in island_ips:
+                    if ipaddress.ip_address(unicode(ip)) in subnet:
+                        found_good_ip = True
+                        break
+                if found_good_ip:
+                    break
+            if not found_good_ip:
+                issues.append(
+                    {'type': 'cross_segment', 'machine': monkey['hostname'],
+                     'networks': [str(subnet) for subnet in monkey_subnets],
+                     'server_networks': [str(subnet) for subnet in get_subnets()]}
+                )
+
+        return issues
+
+    @staticmethod
+    def get_issues():
+        issues = ReportService.get_exploits() + ReportService.get_tunnels() + ReportService.get_cross_segment_issues()
+        issues_dict = {}
+        for issue in issues:
+            machine = issue['machine']
+            if machine not in issues_dict:
+                issues_dict[machine] = []
```

Comment:
get with default value instead of the if block

Reply:
fixed